### PR TITLE
add reference name to resource reference API proto

### PR DESCRIPTION
**Pull request by IronPan**
fix https://github.com/kubeflow/pipelines/issues/1773

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1781)

### CODE REVIEW

File: backend/src/apiserver/model/resource_reference.go [hunk 1]
```diff
@@ -27,6 +27,9 @@ type ResourceReference struct {
 	// The ID of the resource that been referenced to.
 	ReferenceUUID string `gorm:"column:ReferenceUUID; not null; "`
 
+	// The name of the resource that been referenced to.
+	ReferenceName string `gorm:"column:ReferenceName; not null; "`
```
Comment (Ark-kun) [inline backend/src/apiserver/model/resource_reference.go]: Adding a new required field usually breaks backwards compatibility with all existing clients.
What do you think about keeping it optional?

Comment (IronPan) [reply backend/src/apiserver/model/resource_reference.go]: This is database schema requirement which won't be surfaced through UI. Please see integration tests for the sample request.

**Comment by IronPan** [comment]
Verified the travis tests passed. only coverall failed.

File: sdk/python/setup.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
     'google-auth>=1.6.1',
     'requests_toolbelt>=0.8.0',
     'cloudpickle',
-    'kfp-server-api >= 0.1.18, < 0.1.19', #Update the upper version whenever a new version of the kfp-server-api package is released. Update the lower version when there is a breaking change in kfp-server-api.
+    'kfp-server-api >= 0.1.18, <= 0.1.25', #Update the upper version whenever a new version of the kfp-server-api package is released. Update the lower version when there is a breaking change in kfp-server-api.
```
Comment (Ark-kun) [inline sdk/python/setup.py]: It should be `< 0.1.26` so that we can release fix versions like `0.1.25.1`.

**Pull request merged**